index.bs

```diff
@@ -376,42 +408,75 @@ To <dfn>parse an attribution destination</dfn> from a string |str|:
+     : {{AttributionSourceParams/attributionSourcePriority}}
+     :: |anchor|'s <{a/attributionsourcepriority}> attribute
+1. Return the result of running [=obtain an attribution source from params=] with |params|.
+
+<h3 algorithm id="obtaining-event-attribution-source-anchor">Obtaining an event attribution source from an <code>a</code> element</h3>
+
+To <dfn>maybe obtain an event attribution source from an anchor</dfn> given an <{a}> element |anchor|, run the following steps:
+
+1. If |anchor| does not have a <{a/registerattributionsource}> attribute, return null.
+1. Let |source| be the result of running [=obtain an attribution source from an anchor=] with |anchor|.
+1. If |source| is null, return null.
+1. Return the result of running [=obtain an event attribution source from a source=] with |source|.
+
+<h3 algorithm id="obtaining-event-attribution-source-source">Obtaining an event attribution source from a source</h3>
+
+To <dfn>obtain an event attribution source from a source</dfn> given an [=attribution source=] |source|, run the following steps:
+
+1. Set |source|'s [=attribution source/source type=] to "<code>event</code>".
+1. Round |source|'s [=attribution source/expiry=] away from zero to the nearest day.
+1. Return |source|.
+
+<h3 algorithm id="obtaining-attribution-source-params">Obtaining an attribution source from {{AttributionSourceParams}}</h3>
+
+To <dfn>obtain an attribution source from params</dfn> given an {{AttributionSourceParams}} |params|, run the following steps:
+
 1. Let |sourceIdentifier| be a new unique opaque string.
 1. Let |currentTime| be the current time.
-1. If |anchor| does not have both an <{a/attributiondestination}> attribute and
-    an <{a/attributionsourceeventid}> attribute, return null.
-1. If |anchor|'s [=relevant settings object=]'s [=environment settings object/responsible document=] is
+1. If |params| does not have both an {{AttributionSourceParams/attributionDestination}} field and
+    an {{AttributionSourceParams/attributionSourceEventId}} field, return null.
+1. If |params|'s [=relevant settings object=]'s [=environment settings object/responsible document=] is
     not [=allowed to use=] the [=attribution-reporting=] [=policy-controlled feature=], return null.
 1. Let |attributionDestination| be the result of running
-    [=parse an attribution destination=] with |anchor|'s
-    <{a/attributiondestination}> attribute.
+    [=parse an attribution destination=] with |params|'s
+    {{AttributionSourceParams/attributionDestination}} field.
 1. If |attributionDestination| is null, return null.
-1. Let |sourceOrigin| be |anchor|'s [=relevant settings object=]'s
+1. Let |sourceOrigin| be |params|'s [=relevant settings object=]'s
     [=environment/top-level origin=].
 1. Let |reportingOrigin| be |sourceOrigin|.
-1. If |anchor| has an <{a/attributionreportto}> attribute, then:
+1. If |params| has an {{AttributionSourceParams/attributionReportTo}} field, then:
```

Comment:
This can be

```suggestion
1. If |params|["{{AttributionSourceParams/attributionReportTo}}"] [=map/exists=], then:
```

Like the example at https://infra.spec.whatwg.org/#example-map-get